src/parser/clauses/return.c

```diff
@@ -97,23 +62,39 @@ int ReturnClause_ContainsAggregation(const AST_ReturnNode *returnNode) {
 
 void ReturnClause_ReferredEntities(const AST_ReturnNode *returnNode, TrieMap *referred_nodes) {
     if(!returnNode) return;
-    
+
     uint elemCount = array_len(returnNode->returnElements);
     for (uint i = 0; i < elemCount; i++) {
-		AST_ReturnElementNode *retElem = returnNode->returnElements[i];
-        
+        AST_ReturnElementNode *retElem = returnNode->returnElements[i];
         AST_ArithmeticExpressionNode *exp = retElem->exp;
         if(exp) AR_EXP_GetAliases(exp, referred_nodes);
     }
 }
 
+// Collect aggregate functions invoked by the return clause.
+int ReturnClause_AggregateFunctions(const AST_ReturnNode *returnNode, AST_ArithmeticExpressionOP **funcs) {
```

Comment:
I think this function can be removed if we will go with the suggestion regarding functions specifying their input/output types.